tests/python/unittest/test_operator.py

```diff
@@ -1527,7 +1526,7 @@ def check_batchnorm_training(stype):
                 test = mx.symbol.BatchNorm(data, fix_gamma=False, use_global_stats=True, axis=chaxis)
                 check_numeric_gradient(test, in_location, xmean_std, numeric_eps=1e-2, rtol=0.2, atol=0.01)
 
-    stypes = ['row_sparse', 'default']
```

Comment:
I'm just curious. does row_sparse make the test flaky?

Reply:
row_sparse makes it more flaky, and will fail at this part: [testing varying channel axis](https://github.com/apache/incubator-mxnet/blob/e870890ab93ae25d51da01a445c965e4db71069b/tests/python/unittest/test_operator.py#L1524). Which is not in your test_mkddnn:test_batchnorm.

Reply:
Without row_sprase, the flakyness come from difference between gradient values of mkldnn/cuddn implementation of batchnorm and numeric calculation from check_numeric_gradients

Reply:
But isn't that an actual issue? It means that the two implementations are not equivalent and yield different results, right?